includes/Checker/Checks/File_Type_Check.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks\File_Type_Check
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker\Checks;
+
+use Exception;
+use WordPress\Plugin_Check\Checker\Check_Result;
+
+/**
+ * Check to detect disallowed file types.
+ *
+ * @since n.e.x.t
+ */
+class File_Type_Check extends Abstract_File_Check {
+
+	const TYPE_COMPRESSED = 1;
+	const TYPE_PHAR       = 2;
+	const TYPE_VCS        = 4;
+	const TYPE_HIDDEN     = 8;
+	const TYPE_ALL        = 15; // Same as all of the above with bitwise OR.
+
+	/**
+	 * Bitwise flags to control check behavior.
+	 *
+	 * @since n.e.x.t
+	 * @var int
+	 */
+	protected $flags = 0;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $flags Bitwise flags to control check behavior.
+	 */
+	public function __construct( $flags = self::TYPE_ALL ) {
+		$this->flags = $flags;
+	}
+
+	/**
+	 * Amends the given result by running the check on the given list of files.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 * @param array        $files  List of absolute file paths.
+	 *
+	 * @throws Exception Thrown when the check fails with a critical error (unrelated to any errors detected as part of
+	 *                   the check).
+	 */
+	protected function check_files( Check_Result $result, array $files ) {
+		if ( $this->flags & self::TYPE_COMPRESSED ) {
+			$this->look_for_compressed_files( $result, $files );
```

Comment:
These 4 methods seem like they do the same thing. Could we make this more generic like this. 
```
$this->look_for_files( $result, $files, array( 'zip', 'gz', 'tgz', 'rar', 'tar', '7z' ),  'Compressed files are not permitted.', 'compressed_files' );
```

Reply:
Will update.